drivers/hmis/app/models/hmis/hud/processors/base.rb

```diff
@@ -151,32 +151,43 @@ def process_custom_field(field, value)
     }
     # Normalize the input value
     value = normalize_custom_field_value(cded, value)
+
+    # `file` is the only field type that saves an additional record, so do some special-case processing
+    value = process_files(cded, value) if cded.field_type == 'file'
+
     # Infer the field name on the CustomDataElement
     value_field_name = "value_#{cded.field_type}"
 
     # Existing CustomDataElement records for this record with this definition
-    existing_values = existing_custom_data_elements[cded.id] || []
+    existing_cdes = existing_custom_data_elements[cded.id] || []
 
     # If this custom field only allows 1 value and there already is one, update it.
-    if !cded.repeats && existing_values.any?
-      cde_attributes = { id: existing_values.first.id }
+    if !cded.repeats && existing_cdes.any?
+      cde_attributes = { id: existing_cdes.first.id }
       if value.nil?
         cde_attributes[:_destroy] = 1
       else
         cde_attributes[value_field_name] = value
         cde_attributes.merge!(attrs)
       end
-    # If value(s) haven't changed, just update the User and timestamps
-    elsif existing_values.map(&:value) == Array.wrap(value)
-      cde_attributes = existing_values.map { |cde| { id: cde.id, user: @processor.hud_user } }
-    # Else create new custom field value(s), and delete any existing ones.
     else
+      submitted = Array.wrap(value)
```

Comment:
Currently, when any CDE value changes, we delete and replace all of them. See [regression test](https://github.com/greenriver/hmis-warehouse/pull/4910/files#diff-1d09bfc99a44479d26529c3d5ef39deb9f1e365bcb470c9984dd77923fef66daR1804-R1822) farther down on this PR. I think this is a bug, but until now it wouldn't have had any impact except that such CDEs would have incorrect creation timestamps. The reason to fix it now is to prevent deletion and re-creation of File records.

Reply:
Ahh yep, thank you for addressing that!
How will it handle situations where there are repeated values that exist or are submitted? Like `[{value_string: "foo"}, {value_string: "foo"}]` (I'm thinking on an open-choice field, perhaps).
I don't think it's really necessary to preserve that, but it shouldn't explode or do something weird- maybe worth having a test case.

Reply:
I added a test case. It doesn't explode but it does do something weird - see comment below https://github.com/greenriver/hmis-warehouse/pull/4910/files#r1876005116